### DISCUSSION TITLE
HTTP Status code assertion

### DISCUSSION THREAD

**Discussion by abdurraheemj** [seleniumbase/SeleniumBase#1040] [Q&A] [answered]
Hi Michael,
Again me. How do i achieve Status code assertion in SeleniumBase. If you have example. Kindly share it

**Comment by mdmintz** [comment, marked as answer]
Hi @abdurraheemj There are several SeleniumBase methods that you could use for this:

```python
self.get_link_status_code(link)
self.assert_link_status_code_is_not_404(link)
self.assert_no_404_errors()  # For all links on the current page
self.print_unique_links_with_status_codes()  # For current page
self.assert_equal(first, second)
```

**Reply by abdurraheemj** [reply]
Hey Michael, Thanks for the response. On this, self.get_link_status_code(link). if i feed any link which isn't production it throws 404, which is wrong status code.
example: https://www.test.yahoo.com -404; if it is just https://www.test.yahoo.com gives me 200. 
any solution for this kind of problem?

**Reply by mdmintz** [reply]
Both of those are the same URL, and it should return a 404 because the page doesn't exist.

**Reply by abdurraheemj** [reply]
oops..sorry michael. the one returns 200 for https://www.yahoo.com
just for example

**Reply by mdmintz** [reply]
Then it's working correctly. https://www.test.yahoo.com/ should return a 404 because there's no site there. And https://www.yahoo.com should return a 200 because it's a valid site.

**Reply by abdurraheemj** [reply]
i mentioned https://www.test.yahoo.com as an example. Just for confidentiality i am not mentioning the site name which i am working on

**Reply by mdmintz** [reply]
Then it's back to these methods:

```python
self.get_link_status_code(link)
self.assert_link_status_code_is_not_404(link)
self.assert_no_404_errors()  # For all links on the current page
self.print_unique_links_with_status_codes()  # For current page
self.assert_equal(first, second)
```

I'm not sure what exact URLs you are testing against, but the methods that check status codes are using the Python ``requests`` library, which I did not write.

```python
import requests

def _get_link_status_code(link, allow_redirects=False, timeout=5):
    """Get the status code of a link.
    If the timeout is exceeded, will return a 404.
    For a list of available status codes, see:
    https://en.wikipedia.org/wiki/List_of_HTTP_status_codes
    """
    status_code = None
    try:
        response = requests.get(
            link, allow_redirects=allow_redirects, timeout=timeout
        )
        status_code = response.status_code
    except Exception:
        status_code = 404
    return status_code
```
